benchmarks/src/main/java/io/druid/benchmark/query/GroupByBenchmark.java

```diff
@@ -0,0 +1,343 @@
+
+  private IncrementalIndex incIndex;
+  private QueryableIndex qIndex;
+
+  private QueryRunnerFactory factory;
+
+  public static List<AggregatorFactory> QUERY_AGGS;
+  static {
+    QUERY_AGGS = new ArrayList<>();
+    QUERY_AGGS.add(new LongSumAggregatorFactory(
+        "sumA",
+        "sumA"
+    ));
+  }
+
+  static {
+    JSON_MAPPER = new DefaultObjectMapper();
+    INDEX_IO = new IndexIO(
+        JSON_MAPPER,
+        new ColumnConfig()
+        {
+          @Override
+          public int columnCacheSizeBytes()
+          {
+            return 0;
+          }
+        }
+    );
+    INDEX_MERGER_V9 = new IndexMergerV9(JSON_MAPPER, INDEX_IO);
+  }
+
+  @Setup
+  public void setup() throws IOException
+  {
+    if (ComplexMetrics.getSerdeForType("hyperUnique") == null) {
+      ComplexMetrics.registerSerde("hyperUnique", new HyperUniquesSerde(Hashing.murmur3_128()));
+    }
+
+    BenchmarkDataGenerator gen = new BenchmarkDataGenerator(
+        QueryBenchmarkUtil.SCHEMAS,
+        RNG_SEED,
+        QUERY_INTERVAL,
+        MAX_ROWS
+    );
+
+    System.out.printf("SETUP CALLED AT " + +System.currentTimeMillis());
+    incIndex = makeIncIndex();
+
+    for (int j = 0; j < MAX_ROWS; j++) {
+      InputRow row = gen.nextRow();
+      if (j % 10000 == 0) {
+        System.out.println(j + " rows generated.");
+      }
+      incIndex.add(row);
+    }
+
+    File tmpFile = File.createTempFile("GroupByBenchmark-" + System.currentTimeMillis(), ".TEMPDIR");
+    tmpFile.delete();
+    tmpFile.mkdirs();
+    System.out.println(tmpFile.getAbsolutePath() + " isFile: " + tmpFile.isFile() + " isDir:" + tmpFile.isDirectory());
+    tmpFile.deleteOnExit();
+
+    File indexFile = INDEX_MERGER_V9.persist(
+        incIndex,
+        tmpFile,
+        new IndexSpec()
+    );
+    qIndex = INDEX_IO.loadIndex(indexFile);
+
+    OffheapBufferPool bufferPool = new OffheapBufferPool(250000000, Integer.MAX_VALUE);
+    OffheapBufferPool bufferPool2 = new OffheapBufferPool(250000000, Integer.MAX_VALUE);
+    final GroupByQueryConfig config = new GroupByQueryConfig();
+    config.setMaxIntermediateRows(1000000);
+    config.setMaxResults(1000000);
+
+    final Supplier<GroupByQueryConfig> configSupplier = Suppliers.ofInstance(config);
+    final GroupByQueryEngine engine = new GroupByQueryEngine(configSupplier, bufferPool);
+
+    factory = new GroupByQueryRunnerFactory(
+        engine,
+        QueryBenchmarkUtil.NOOP_QUERYWATCHER,
+        configSupplier,
+        new GroupByQueryQueryToolChest(
+            configSupplier, JSON_MAPPER, engine, bufferPool2,
+            QueryBenchmarkUtil.NoopIntervalChunkingQueryRunnerDecorator()
+        ),
+        bufferPool2
+    );
+  }
+
+  @Setup(Level.Iteration)
+  public void setup2() throws IOException
+  {
+
+  }
+
+  @TearDown(Level.Iteration)
+  public void teardown() throws IOException
+  {
+  }
+
+  private IncrementalIndex makeIncIndex()
+  {
+    return new OnheapIncrementalIndex(
+        0,
```

Comment:
[minor] I kinda prefer the IncrementalIndexSchema based constructor, the builder makes it easier to read what's going on.